silx/gui/plot/items/roi.py

```diff
@@ -395,229 +363,283 @@ def setFirstShapePoints(self, points):
+            value = source.getName()
+            for d in destination:
+                d.setName(value)
+        elif event == items.ItemChangedType.EDITABLE:
+            value = source.isEditable()
+            for d in destination:
+                d.setEditable(value)
+        elif event == items.ItemChangedType.SELECTABLE:
+            value = source.isSelectable()
+            for d in destination:
+                d._setSelectable(value)
+        elif event == items.ItemChangedType.COLOR:
+            value = rgba(source.getColor())
+            for d in destination:
+                d.setColor(value)
+        elif event == items.ItemChangedType.LINE_STYLE:
+            value = self.getLineStyle()
+            for d in destination:
+                d.setLineStyle(value)
+        elif event == items.ItemChangedType.LINE_WIDTH:
+            value = self.getLineWidth()
+            for d in destination:
+                d.setLineWidth(value)
+        elif event == items.ItemChangedType.SYMBOL:
+            value = self.getSymbol()
+            for d in destination:
+                d.setSymbol(value)
+        elif event == items.ItemChangedType.SYMBOL_SIZE:
+            value = self.getSymbolSize()
+            for d in destination:
+                d.setSymbolSize(value)
+        elif event == items.ItemChangedType.VISIBLE:
+            value = self.isVisible()
+            for d in destination:
+                d.setVisible(value)
+        else:
+            assert False
 
-        if not nbPointsChanged and numpy.allclose(points, self._points,
-                                                  rtol=0, atol=0, equal_nan=True):
-            # There is no changes
-            return
+    def _editingStarted(self):
+        assert self._editable is True
+        self.sigEditingStarted.emit()
 
-        self._points = points
+    def _editingFinished(self):
+        self.sigEditingFinished.emit()
 
-        self._updateShape()
-        if self._items and not nbPointsChanged:  # Update plot items
-            item = self._getLabelItem()
-            if item is not None:
-                markerPos = self._getLabelPosition()
-                item.setPosition(*markerPos)
-
-            if self._editAnchors:  # Update anchors
-                for anchor, point in zip(self._editAnchors, points):
-                    old = anchor.blockSignals(True)
-                    anchor.setPosition(*point)
-                    anchor.blockSignals(old)
-
-        else:  # No items or new point added
-            # re-create plot items
-            self._createPlotItems()
 
-        self.sigRegionChanged.emit()
+class _Foo:
+    """This inheritance is needed to avoid wrong __init__ order in Qt"""
+    pass
 
-    def _updateShape(self):
-        """Called when shape must be updated.
 
-        Must be reimplemented if a shape item have to be updated.
-        """
-        return
+class _HandleBasedROI(RegionOfInterest, _Foo):
```

Comment:
Why do you get the _Foo class? This is weird.

Reply:
It was fixes the way the inheritance was behaving, removing a weird exception. I dont have solution here.

Reply:
I could try to rebase and take a look again.

Reply:
You can rebase as I merged PR #3034 which should fix CI.
I'll have a look at this _Foo now.

Reply:
Only alternative so far is to constraint the order of inheritance further down which is also kind of awkward.... so let's see that later and keep the ugly workaround at a single place rather than forcing inheritance order at mainy places.